File: C/20_Divide&Conquer/BOJ_05_11401_이항계수3.c
#include <stdio.h>
const int p = 1000000007;
long long int pow(int a, int b) {
	if (b == 1) return a % p;
	long long int tmp = pow(a, b / 2);
	if (b % 2 == 1) {
		return (tmp * tmp % p) * a % p;
	}
	return tmp * tmp % p;
}
long long int factorial(long long int n) {
	if (n <= 1) return 1;
	else return n * factorial(n - 1) % p;
}
main() {
	int n, k;
	scanf("%d %d", &n, &k);
	long long int numerator = factorial(n);
	long long denominator = factorial(k) * factorial(n - k) % p;
	printf("%d", (numerator * pow(denominator, p - 2)) % p);
}

File: JAVA/20_Divide&Conquer/BOJ_05_11401_이항계수3.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    final static int p = 1000000007;

    public static long pow(long a, int b) {
        if (b == 1) {
            return a % p;
        }
        long tmp = pow(a, b / 2);
        if (b % 2 == 1) {
            return (tmp * tmp % p) * a % p;
        }
        return tmp * tmp % p;
    }

    public static long factorial(long n) {
        if (n <= 1) return 1;
        else return n * factorial(n - 1) % p;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());
        long numerator = factorial(n) % p;
        long denominator = factorial(k) * factorial(n - k) % p;
        System.out.println((numerator * pow(denominator, p - 2)) % p);
    }
}